cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -870,11 +870,18 @@ func (a *StaticAutoscaler) deleteCreatedNodesWithErrors() bool {
 			klog.Warningf("Error while trying to delete nodes from %v: %v", nodeGroupId, err)
 		} else {
 			deletedAny = true
-			a.clusterStateRegistry.InvalidateNodeInstancesCacheEntry(nodeGroup)
+			err = a.clusterStateRegistry.HandleDeletingFailedScaleUpNodes(nodeGroup)
+			if err != nil {
+				returnErr = fmt.Errorf("Failed to clean up state after deleting failed nodes from %s node group: %v", nodeGroup.Id(), err)
+				klog.Warningf(returnErr.Error())
+			}
 		}
 	}
 
-	return deletedAny
+	if deletedAny {
+		klog.V(0).Infof("Some nodes that failed to create were removed.")
```

Comment:
nit: I wouldn't mind logging this per-nodegroup and getting rid of this variable entirely. I think it is an important enough message to be worth logging with details and any scenario where log volume of this becomes a problem is very scary to imagine.
Feel free to ignore this comment if you disagree - especially if you decide to call csr.Recalculate() in this branch as per my other comment.

Reply:
I went with recalculating here, so I left it as is